### PR TITLE
Test if mode is unique

### PR DESCRIPTION
**Pull request by ocefpaf**
@lsetiawan ignore this for a while. I am just testing to see if I can get it to fail.

### CODE REVIEW

File: wof/wofpy_config.py [hunk 1]
```diff
@@ -89,5 +95,9 @@ def main():
         copytree(_CONFIG, _PRODUCTION, overwrite=overwrite)
 
 
+def main():
+    args = parse_args()
+    cli(args)
```
Comment (ocefpaf) [inline wof/wofpy_config.py]: Factoring the `cli` out of `main` helps us to create "real user case" tests.

File: test/test_cli.py [hunk 2]
```diff
@@ -32,10 +32,16 @@ def test_makedirs_overwrite_soft():
...
+    args = {
+        'INDIR': _directory,
+        '--mode': 'production',
+        '--overwrite': 'hard'
+    }
+    cli(args)
+    assert os.path.isdir(os.path.join(_directory, 'production_configs'))
+    args.update({'--mode': 'development'})
+    cli(args)
+    assert not os.path.isdir(os.path.join(_directory, 'production_configs'))
```
Comment (ocefpaf) [inline test/test_cli.py]: This sequence of `assert`s check if an instance created under `production` mode lost the `production_configs` when it was overwritten by a second call with `development`. This is not an ideal test b/c it test a corner case only, but it works on Windows and we can add additional cases later. (For example, a test that modifies the contents of the config files and see if the expected modification are there, or not, when calling the `cli`).

**Comment by lsetiawan** [comment]
> I don't know why Travis-CI recycles the id_node, but the directory is overwritten. Probably some smart disk cache? Dunno...

Wow. So many changes on Travis CI side! I guess it keeps us on our toes 😄

**Comment by emiliom** [comment]
Thanks.
